[export] nested terms in nn_module_stack deserialization

pianpwk opened the pull request:
Summary: accounting for terms like "getattr(getattr(a[0], b), c)".

Test Plan: test_serialize

Differential Revision: D68784736

angelayi commented on torch/_export/serde/serialize.py:
```diff
@@ -2455,17 +2455,22 @@ def deserialize_meta_func(serialized_target: str):
+                        start = end + 1
+                out.append(metadata[start:])
+                assert len(out) == 3
+                return out
```
could we add a test case for the failure that required this fix?

pianpwk committed:
[export] nested terms in nn_module_stack deserialization (#145901)

Summary:

accounting for terms like "getattr(getattr(a[0], b), c)".

Test Plan: test_serialize

Reviewed By: angelayi

Differential Revision: D68784736
